rfd/0033-sni-alpn-teleport-proxy-routing.md

```diff
@@ -0,0 +1,160 @@
+                                                                                                  
+                                                      ┌───────────────┐                           
+                                                      │Teleport Proxy │                           
+                                                      │   Listener    │                           
+                                                      │               │                           
+                                                      └───────────────┘                           
+                                                              │                                   
+                                                              │                                   
+                                                              │                                   
+                                                              │                                   
+                                            ┌─────────────────┴─────────────────┐                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            ▼                                   ▼                 
+                             ┌────────────────────────────┐    ┌─────────────────────────────────┐
+                             │SNI:                        │    │SNI:                             │
+                             │  proxy.example.teleport.com│    │ kube.proxy.example.teleport.com │
+                             │  default                   │    │                                 │
+                             └────────────────────────────┘    └─────────────────────────────────┘
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                   │                 
+                                            │                                 ALPN:               
+                                            │                         http1.1, h2, default        
+           ┌────────────────┬───────────────┴──┬───────────────┐                │                 
```

Comment:
Can we add auth service here as well? This will allow clients to talk to GRPC endpoint directly

Reply:
Yes, The teleport-auth allows to dial auth service directly without SSH Tunnels.